src/main/kotlin/org/jetbrains/numkt/NativeLibUtils.kt

```diff
@@ -16,141 +16,102 @@
 
 package org.jetbrains.numkt
 
-import java.io.BufferedReader
 import java.io.File
 import java.io.FileNotFoundException
-import java.io.InputStreamReader
 import java.nio.file.Files
-import java.util.jar.JarFile
+
+private const val namePythonScript = "utils"
+private const val baseNameNativeLib = "ktnumpy"
 
 enum class OSType {
     MACOS, LINUX, WINDOWS, UNKNOWN
 }
 
-enum class ArchType {
-    X86_64, ARM, UNKNOWN
-}
-
-fun loadLibrary(baseName: String) {
-
-    val exception: Throwable
-    try {
-        System.loadLibrary(baseName)
-        return
-    } catch (e: Throwable) {
-        exception = e
-    }
-
-    try {
-        load(baseName)
-    } catch (e: Throwable) {
-        e.printStackTrace()
-        exception.printStackTrace()
-    }
-}
-
-fun load(baseName: String) {
-    val libName = System.mapLibraryName(baseName)
-    val osName = getOS()
-
-    val libDest = File(
-        when (osName) {
-            OSType.MACOS -> "/usr/local/lib/$libName"
-            OSType.LINUX -> "/usr/local/lib/$libName"
-            OSType.WINDOWS -> "C:\\System32\\$libName"
-            else -> "unknown"
-        }
-    )
-
-    if (libDest.exists()) {
-        System.load(libDest.absolutePath)
-        return
+data class PythonConf(val osType: OSType, val pythonHome: String, val pythonLibPath: String)
+
+object LibraryLoader {
+    private val nameNativeLib = System.mapLibraryName(baseNameNativeLib)
+    private val tmpDir: File //= Files.createTempDirectory("nativeKtNumPy").toFile().apply { deleteOnExit() }
+    val pythonConf by lazy {
+        val osType = getOS()
+        val pythonHome = System.getenv("PYTHONHOME") ?: getPythonEnv(namePythonScript, "get_python_home")
+        PythonConf(
+            osType = osType,
+            pythonHome = pythonHome,
+            pythonLibPath = if (osType == OSType.WINDOWS) buildString {
+                append(pythonHome)
+                append('\\')
+                append(getPythonEnv(namePythonScript, "get_pylib"))
+                append(".dll")
+            } else getPythonEnv(
+                namePythonScript,
+                "get_pylib_path"
+            )
+        )
     }
 
-    val nativeLibraryPath = buildString {
-        append("META-INF/Native/")
-        when (osName) {
-            OSType.MACOS -> append("macos")
-            OSType.LINUX -> append("linux")
-            OSType.WINDOWS -> append("win")
-            else -> {
+    init {
+        val dirScr = File("buildScr/python")
+        if (dirScr.exists()) {
+            tmpDir = dirScr
+        } else {
+            tmpDir = Files.createTempDirectory("nativeKtNumPy").toFile().apply { deleteOnExit() }
```

Comment:
[Optional] May be changed to `apply(File::deleteOnExit)` to avoid lambda creation